source/Sashimi/AzureWebAppHealthCheckActionHandler.cs

```diff
@@ -26,4 +26,25 @@ public IActionHandlerResult Execute(IActionHandlerContext context, ITaskLog task
                           .Execute(taskLog);
         }
     }
+
+    class AzureWebAppDiscoveryActionHandler : IActionHandler
+    {
+        static readonly CalamariFlavour CalamariAzureAppService = new CalamariFlavour("Calamari.AzureAppService");
+
+        public string Id => SpecialVariables.Action.Azure.WebAppHealthCheckActionTypeName;
+        public string Name => "Discover Azure Web Apps";
+        public string Description => "Discover available Azure Web Apps.";
+        public string? Keywords => null;
+        public bool ShowInStepTemplatePickerUI => false;
+        public bool WhenInAChildStepRunInTheContextOfTheTargetMachine => false;
+        public bool CanRunOnDeploymentTarget => false;
+        public ActionHandlerCategory[] Categories => new[] { ActionHandlerCategory.BuiltInStep, ActionHandlerCategory.Azure };
+
+        public IActionHandlerResult Execute(IActionHandlerContext context, ITaskLog taskLog)
+        {
+            return context.CalamariCommand(CalamariAzureAppService, "target-discovery")
+              .WithCheckAccountIsNotManagementCertificate(context, taskLog)
```

Comment:
When I was testing and had this line in it was producing a warning even though the account for discovery was a service principal, I think it might be because it checks for a known variable which doesn't exist in the discovery property bag?

Reply:
Yes, you're right. It is checking the variable `Octopus.Account.AccountType`, so won't work here.
I think the whole check is redundant anyway, since it is a warning of a deprecation in 2018!
I've removed it here.